src/main/java/com/pubnub/api/managers/ReconnectionManager.java

```diff
@@ -95,16 +109,26 @@ int getNextInterval() {
             } else if (timerInterval < 1) {
                 timerInterval = MIN_EXPONENTIAL_BACKOFF;
             }
-            log.debug("timerInterval = " + timerInterval + " at: " + Calendar.getInstance().getTime().toString());
+            timerInterval = (int) ((timerInterval + getRandomDelay()) * MILLISECONDS);
+            log.debug("timerInterval = " + timerInterval + "ms at: " + Calendar.getInstance().getTime().toString());
         }
 
         if (pnReconnectionPolicy == PNReconnectionPolicy.LINEAR) {
-            timerInterval = LINEAR_INTERVAL;
+            timerInterval = (int) ((LINEAR_INTERVAL + getRandomDelay()) * MILLISECONDS);
         }
-
         return timerInterval;
     }
 
+    private double getRandomDelay() {
+        double randomDelay = MAX_RANDOM * random.nextDouble();
+        return roundTo3DecimalPlaces(randomDelay);
+    }
+
+    private double roundTo3DecimalPlaces(double value) {
```

Comment:
ditto, don't use text formatting for manipulating numbers

Reply:
ok